owa/app/js/components/manage_patient_identifier_sources/edit_patient_identifier_sources/localIdentifierGenerator.jsx

```diff
@@ -0,0 +1,144 @@
+      initialized: props.identifierSource.initialized,
+      uuid: props.identifierSource.uuid,
+      name: props.identifierSource.name,
+      description: props.identifierSource.description,
+      identifierTypeValidator: props.identifierSource.identifierTypeValidator,
+      baseCharacterSet: props.identifierSource.baseCharacterSet,
+      firstIdentifierBase: props.identifierSource.firstIdentifierBase,
+      prefix: props.identifierSource.prefix,
+      suffix: props.identifierSource.suffix,
+      minLength: props.identifierSource.minLength,
+      maxLength: props.identifierSource.maxLength
+    }
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  handleFirstIdentifierBase(){
+    if(this.state.initialized){
+      return(this.state.firstIdentifierBase + " In Use - Unable to Modify ")
+    }
+    return(<Input 
+    id="firstIdentifierBase" 
+    value={this.state.firstIdentifierBase == null ? "" : this.state.firstIdentifierBase} 
+    onChange={event => this.handleChange(event)} 
+    name="firstIdentifierBase" 
+    className="editPatientIdentifierSourcesInput" 
+    size="sm" 
+    type="text" />);
+  }
+
+  handleChange(event){
+    this.setState({
+      [event.target.name]: event.target.value
+    });
+  }
+
+  handleSubmit(){
+    let data = {
+      name: this.state.name,
+      description: this.state.description,
+      baseCharacterSet: this.state.baseCharacterSet,
+      firstIdentifierBase: this.state.firstIdentifierBase,
+      prefix: this.state.prefix,
+      suffix: this.state.suffix,
+      minLength: this.state.minLength,
+      maxLength: this.state.maxLength
+    };
+
+    apiCall(data, 'post', '/idgen/identifiersource/' + this.state.uuid).then((response) => {
+      if( response["error"] === undefined ) {
+        this.props.handleUpdateName(this.state.index, this.state.name);
+        this.props.handleAlerts("success", "Successfull Updated " + this.state.name); 
+      }else{
+        this.props.handleAlerts("error", response["error"]["message"]);
+      }  
+    });
+  }
+
+  render() {
+
+    return (
+      <div>
+        <div className="editPatientIdentifierSourcesTitle">Edit: Local Identifier Generator for {this.props.identifierSource.identifierTypeName}</div>
+        <Table size="sm" responsive>
+          <thead>
+            <tr>
+              <td className="editPatientIdentifierSourcesTitle">Field</td>
+              <td className="editPatientIdentifierSourcesTitle">Value</td>
+            </tr>
+          </thead>
+          <tbody>
+            <tr>
+              <td className="editPatientIdentifierSourcesTitle"><span className="requiredField">*</span> Name:</td>
+              <td>
+                <Input id="name" value={this.state.name == null ? "" : this.state.name} onChange={event => this.handleChange(event)} name="name" className="editPatientIdentifierSourcesInput" type="text" size="sm" />
```

Comment:
Same with the onChange event here as well

Reply:
true

Reply:
fixed